src/handlers/v1/markets/[chainId]/[marketId]/zap-borrow/index.ts

```diff
@@ -0,0 +1,153 @@
+  method: 'POST',
+  path: '/v1/markets/{chainId}/{marketId}/zap-borrow',
+  handler: async (event) => {
+    const chainId = Number(event.pathParameters.chainId);
+    const marketId = event.pathParameters.marketId.toUpperCase();
+    if (!validateMarket(chainId, marketId)) {
+      throw newHttpError(400, { code: '400.1', message: 'market does not exist' });
+    }
+
+    if (!event.body) {
+      throw newHttpError(400, { code: '400.2', message: 'body is invalid' });
+    }
+
+    let account = event.body.account;
+    if (!account) {
+      throw newHttpError(400, { code: '400.3', message: `account can't be blank` });
+    }
+    try {
+      account = utils.getAddress(account);
+    } catch {
+      throw newHttpError(400, { code: '400.4', message: 'account is invalid' });
+    }
+
+    const service = new Service(chainId);
+
+    let marketInfo: MarketInfo;
+    try {
+      marketInfo = await service.getMarketInfo(marketId, account);
+    } catch (err) {
+      throw newInternalServerError(err);
+    }
+    const { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR } = marketInfo;
+    const currentPosition = { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR };
+
+    let targetTokenAmount = '0';
+    const logics: GetZapBorrowQuotationResponseBody['logics'] = [];
+    let approvals: GetZapBorrowQuotationResponseBody['approvals'] = [];
+    let targetPosition = currentPosition;
+    if (event.body.amount && event.body.targetToken && Number(event.body.amount) > 0) {
+      const { amount, slippage } = event.body;
+      const {
+        baseToken,
+        supplyAPR,
+        supplyUSD,
+        borrowAPR,
+        borrowCapacityUSD,
+        availableToBorrow,
+        liquidationLimit,
+        baseTokenPrice,
+      } = marketInfo;
+      const targetToken = common.Token.from(event.body.targetToken);
+
+      // 1. check supply USD
+      if (!new BigNumberJS(supplyUSD).isZero()) {
+        throw newHttpError(400, { code: '400.5', message: 'supply USD is not zero' });
+      }
+
+      // 2. check borrow amount
+      if (new BigNumberJS(amount).gt(new BigNumberJS(availableToBorrow))) {
+        throw newHttpError(400, { code: '400.6', message: 'borrow amount is greater than available amount' });
+      }
+
+      // 3. new and append compound v3 borrow logic
+      logics.push(
+        apisdk.protocols.compoundv3.newBorrowLogic({
+          marketId,
+          output: {
+            token: !targetToken.is(baseToken.wrapped) && targetToken.is(baseToken) ? baseToken : baseToken.wrapped,
+            amount,
+          },
+        })
+      );
+
+      // 4. new and append swap token logic
+      if (targetToken.unwrapped.is(baseToken)) {
+        targetTokenAmount = amount;
+      } else {
+        const quotation = await apisdk.protocols.paraswapv5.getSwapTokenQuotation(chainId, {
+          input: { token: baseToken.wrapped, amount },
```

Comment:
I think there's no need to do `wrapped` here?

Reply:
Considering `baseToken` and `targetToken` are not the same regardless of the wrap/unwrap, `baseToken.wrapped` is expected to be borrowed from Compound V3.